React app with styled components is missing a file

### Current Behavior

This is happening on a freshly created monorepo, with one React app configured by the generator for Styled Components.

```
❯ npx nx serve web

> nx run web:serve

<i> [webpack-dev-server] Project is running at:
<i> [webpack-dev-server] Loopback: http://localhost:4200/, http://127.0.0.1:4200/
<i> [webpack-dev-server] 404s will fallback to '/index.html'
●  ━━━━━━━━━━━━━━━━━━━━━━━━━ (98%) emitting emit                                                                                                                                                                                                                                                                    
 ERROR  Compilation failed. See above for more details.

asset main.js 1.54 MiB [emitted] (name: main)
asset 3rdpartylicenses.txt 28.5 KiB [emitted]
asset favicon.ico 14.7 KiB [emitted]
asset index.html 346 bytes [emitted]
asset assets/.gitkeep 0 bytes [emitted]
runtime modules 31.9 KiB 17 modules
modules by path ../../node_modules/ 1.38 MiB 51 modules
modules by path ./src/ 97.6 KiB
  ./src/main.tsx 975 bytes [built] [code generated] [1 error]
  ./src/app/app.tsx 1.38 KiB [built] [code generated]
  ./src/app/nx-welcome.tsx 95.3 KiB [built] [code generated]

ERROR in ./src/main.tsx 5:0-36
  × Module not found: Can't resolve './styles.styled-components' in '/redacted/apps/web/src'
   ╭─[5:7]
 3 │ import * as ReactDOM from 'react-dom/client';
 4 │ import App from './app/app';
 5 │ import './styles.styled-components';
   ·        ────────────────────────────
 6 │ var root = ReactDOM.createRoot(document.getElementById('root'));
 7 │ root.render(/*#__PURE__*/ _jsxDEV(StrictMode, {
   ╰────


Rspack 1.0.5 compiled with 1 error in 111 ms
```

### Expected Behavior

The freshly created empty app can start

### GitHub Repo

_No response_

### Steps to Reproduce

1. Create new nx workspace
2. Create new react app, choose Styled Components in the generator
3. Try to serve the app

### Nx Report

```
NX   Report complete - copy this into the issue template

Node           : 16.20.2
OS             : darwin-arm64
Native Target  : aarch64-macos
npm            : 8.19.4

nx                 : 20.0.3
@nx/js             : 20.0.3
@nx/jest           : 20.0.3
@nx/eslint         : 20.0.3
@nx/workspace      : 20.0.3
@nx/devkit         : 20.0.3
@nx/eslint-plugin  : 20.0.3
@nx/react          : 20.0.3
@nx/rspack         : 20.0.3
@nx/web            : 20.0.3
typescript         : 5.5.4
---------------------------------------
Registered Plugins:
@nx/eslint/plugin
@nx/jest/plugin
```

### Failure Logs

```
ERROR in ./src/main.tsx 5:0-36
  × Module not found: Can't resolve './styles.styled-components' in '/Users/wiz/Work/swatcher/apps/web/src'
   ╭─[5:7]
 3 │ import * as ReactDOM from 'react-dom/client';
 4 │ import App from './app/app';
 5 │ import './styles.styled-components';
   ·        ────────────────────────────
 6 │ var root = ReactDOM.createRoot(document.getElementById('root'));
 7 │ root.render(/*#__PURE__*/ _jsxDEV(StrictMode, {
```

### Package Manager Version

_No response_

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/next/src/generators/application/files/app/page.tsx__tmpl__
+++ b/packages/next/src/generators/application/files/app/page.tsx__tmpl__
@@ -1,19 +1,16 @@
-<% if (style !== 'none') { %>
-  <% if (styledModule && (styledModule === 'styled-jsx' || styledModule === 'styled-components'))  {%>
-  'use client';
-  <% }%> 
-  <% if (styledModule && styledModule !== 'styled-jsx') {
-    var wrapper = 'StyledPage';
-  %>import styled from '<%= styledModule %>';<% } else {
-    var wrapper = 'div';
-  %>
-    <%- (style !== 'styled-jsx' && style !== 'tailwind') ? `import styles from './page.module.${style}';` : '' %>
-  <% }
-  %>
+<%_ if (style !== 'none') { _%>
+  <%_ if (styledModule && (styledModule === 'styled-jsx' || styledModule === 'styled-components'))  { _%>
+'use client';
+  <%_ } _%> 
+  <%_ if (styledModule && styledModule !== 'styled-jsx') { var wrapper = 'StyledPage'; _%>
+import styled from '<%= styledModule %>';
+<%_ } else { var wrapper = 'div'; _%>
+<%- (style !== 'styled-jsx' && style !== 'tailwind') ? `import styles from './page.module.${style}';` : '' %>
+  <%_ } _%>
 
-  <% if (styledModule && styledModule !== 'styled-jsx') { %>
-  const StyledPage = styled.div`<%- pageStyleContent %>`;
-  <% }%>
+  <%_ if (styledModule && styledModule !== 'styled-jsx') { _%>
+const StyledPage = styled.div`<%- pageStyleContent %>`;
+  <%_ } _%>
 <% } %>
 export default function Index() {
   /*

--- a/packages/next/src/generators/application/files/common/specs/__fileName__.spec.tsx__tmpl__
+++ b/packages/next/src/generators/application/files/common/specs/__fileName__.spec.tsx__tmpl__
@@ -1,14 +1,14 @@
 import React from 'react';
 import { render } from '@testing-library/react';
-<% if (src && !appDir) { %>
+<%_ if (src && !appDir) { _%>
 import Index from '../src/pages/index';
-<% } else if (!appDir) { %>
+<%_ } else if (!appDir) { _%>
 import Index from '../pages/index';
-<% } else if (appDir && src) {%>
+<%_ } else if (appDir && src) { _%>
 import Page from '../src/app/page';
-<% } else { %>
+<%_ } else { _%>
 import Page from '../app/page';
-<% } %>
+<%_ } _%>
 
 describe(<%- appDir ? `'Page'` : `'Index'` %>, () => {
   it('should render successfully', () => {

--- a/packages/next/src/generators/application/files/pages/__fileName__.tsx__tmpl__
+++ b/packages/next/src/generators/application/files/pages/__fileName__.tsx__tmpl__
@@ -1,17 +1,14 @@
-<% if (style !== 'none') { %>
-  <% if (styledModule && styledModule !== 'styled-jsx') {
-    var wrapper = 'StyledPage';
-  %>import styled from '<%= styledModule %>';<% } else {
-    var wrapper = 'div';
-  %>
-    <%- (style !== 'styled-jsx' && style !== 'tailwind') ? `import styles from './${fileName}.module.${style}';` : '' %>
-  <% }
-  %>
+<%_ if (style !== 'none') { _%>
+  <%_ if (styledModule && styledModule !== 'styled-jsx') { var wrapper = 'StyledPage'; _%>
+import styled from '<%= styledModule %>';
+  <% } else { var wrapper = 'div';%>
+<%- (style !== 'styled-jsx' && style !== 'tailwind') ? `import styles from './${fileName}.module.${style}';` : '' %>
+  <%_ } _%>
 
-  <% if (styledModule && styledModule !== 'styled-jsx') { %>
-  const StyledPage = styled.div`<%- pageStyleContent %>`;
-  <% }%>
-<% } %>
+  <%_ if (styledModule && styledModule !== 'styled-jsx') { _%>
+const StyledPage = styled.div`<%- pageStyleContent %>`;
+  <%_ } _%>
+<%_ } _%>
 
 export function Index() {
   /*

--- a/packages/react/src/generators/application/files/base-rspack/src/main.tsx__tmpl__
+++ b/packages/react/src/generators/application/files/base-rspack/src/main.tsx__tmpl__
@@ -1,9 +1,10 @@
 <%_ if (strict) { _%>import { StrictMode } from 'react';<%_ } _%>
 import * as ReactDOM from 'react-dom/client';
 <%_ if (routing) { _%>import { BrowserRouter } from 'react-router-dom';<%_ } _%>
-
 import App from './app/<%= fileName %>';
-import './styles.<%= style %>'
+<%_ if(hasStyleFile) { _%>
+import './styles.<%= style %>' 
+<%_ } _%>
 
 const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
 <%_ if(strict && !routing) { _%>

--- a/packages/react/src/generators/application/lib/create-application-files.ts
+++ b/packages/react/src/generators/application/lib/create-application-files.ts
@@ -40,6 +40,7 @@ export async function createApplicationFiles(
   } else {
     styleSolutionSpecificAppFiles = '../files/style-css-module';
   }
+  const hasStyleFile = ['scss', 'css', 'less'].includes(options.style);
 
   const onBoardingStatus = await createNxCloudOnboardingURLForWelcomeApp(
     host,
@@ -64,6 +65,7 @@ export async function createApplicationFiles(
     appTests,
     inSourceVitestTests: getInSourceVitestTestsTemplate(appTests),
     style: options.style === 'tailwind' ? 'css' : options.style,
+    hasStyleFile,
   };
 
   if (options.bundler === 'vite') {

